hugolib/author.go

```diff
@@ -43,3 +50,84 @@ type Author struct {
 // - linkedin
 // - skype
 type AuthorSocial map[string]string
+
+// URL is a convenience function that provides the correct canonical URL
+// for a specific social network given a username. If an unsupported network
+// is requested, only the username is returned
+func (as AuthorSocial) URL(key string) string {
+	switch key {
+	case "github":
+		return "https://github.com/" + as[key]
+	case "facebook":
+		return "https://www.facebook.com/" + as[key]
+	case "twitter":
+		return "https://twitter.com/" + as[key]
+	case "googleplus":
+		return "https://plus.google.com/+" + as[key]
+	case "pinterest":
+		return "https://www.pinterest.com/" + as[key] + "/"
+	case "instagram":
+		return "https://www.instagram.com/" + as[key] + "/"
+	case "youtube":
+		return "https://www.youtube.com/user/" + as[key]
+	case "linkedin":
+		return "https://www.linkedin.com/in/" + as[key]
+	default:
+		return as[key]
+	}
+}
+
+func mapToAuthorList(m map[string]interface{}) AuthorList {
+	authors := make(AuthorList, len(m))
+	for authorKey, data := range m {
+		authorMap, ok := data.(map[string]interface{})
+		if !ok {
+			continue
+		}
+		authors[authorKey] = mapToAuthor(authorMap)
+	}
+	return authors
+}
+
+func mapToAuthor(m map[string]interface{}) Author {
+	author := Author{}
+	for k, data := range m {
+		switch k {
+		case "given_name":
+			author.GivenName = cast.ToString(data)
```

Comment:
I can't remember for sure, but I think it was an issue with the data files all being lowercase.